doc: samples: Update bluetooth/central README (#87972)
zephyrproject-rtos/zephyr: samples/bluetooth/central/README.rst

   :name: Central
   :relevant-api: bluetooth

   Implement basic Bluetooth LE Central role functionality
   (scanning and connecting).

Overview
********

This application demonstrates basic Bluetooth LE Central role functionality
by scanning for other Bluetooth LE devices and establishing a connection
to the first one with a strong enough signal.

Core features
*************

Scanning for devices
====================

> The application initiates a passive scan to detect nearby BLE devices.

Reviewer: ```suggestion
The application initiates a passive scan to detect nearby BT LE devices.
```

Let's not use the unofficial BLE abbreviation, but rather the official BT LE

Applies multiple places

Reviewer: Alternative stick to the use of `Bluetooth LE` as is otherwise used in this file

Reviewer 2: If we're going to be picky, I don't think "BT" is official either, rather you should use "Bluetooth" :)

Author: changed it to Bluetooth LE

Reviewer: > If we're going to be picky, I don't think "BT" is official either, rather you should use "Bluetooth" :)

That's a good point, and what's worse is that the Bluetooth core spec actually has a BT abbreviation for something else:

> bandwidth-symbol time product (BT) 

:D

It specifically looks for devices that have a signal strength greater

> than -50dBm.This threshold helps the app filter out weaker signals,

Reviewer: ```suggestion
than -50dBm. This threshold helps the app filter out weaker signals,
```

ensuring it only interacts with devices that are within a reasonable

> range for communication.

Author: done

Connection handling
===================

1. The central scans for peripheral devices and if it finds a peripheral

>    which has a signal strength higher than -50dBm, an attempt to establish

Author: done

   LE connection is made.
2. If the connection is successful, the central initiates disconnect to
   the peripheral and then restarts the scan.
3. If there are no connections, the central keeps scanning continuously.

The sample is used to demonstrate the Central mode capabilities of BLE and
hence a disconnect is issued right after connection with a peripheral,

> to keep the central in scan mode.

Author: done

   :name: Central
   :relevant-api: bluetooth

   Implement basic Bluetooth LE Central role functionality
   (scanning and connecting).

> OVERVIEW

Reviewer 3: no capital letters in headings please

Reviewer 3: please let reviewer close conversations as it's otherwise really difficult to "re"-review PRs efficiently

https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#contributor-expectations:~:text=Unless%20they%20applied%20the%20reviewer%E2%80%99s%20recommendation%20exactly%2C%20authors%20must%20not%20resolve%20and%20hide%20comments%2C%20they%20must%20let%20the%20initial%20reviewer%20do%20it.

Author: Sure @kartben

********

> Application demonstrates basic Bluetooth LE Central role functionality

Reviewer 3: ```suggestion
This application demonstrates basic Bluetooth LE Central role functionality
```

by scanning for other Bluetooth LE devices and establishing a connection
to the first one with a strong enough signal.

CORE FEATURES
*************

> 1. Scanning for devices:
> ************************

Reviewer 3: ```suggestion
Scanning for devices
====================

```

Author: Done

The application initiates a passive scan to detect nearby BLE devices.
It specifically looks for devices that have a signal strength greater
than -50dBm.This threshold helps the app filter out weaker signals,
ensuring it only interacts with devices that are within a reasonable
range for communication.

> 2. Connection Handling:
> ***********************

Reviewer 3: ```suggestion
Connection Handling
===================

```

Author: Done

1. The central scans for peripheral devices and if it finds a peripheral
   which has a signal strength higher than -50dBm, an attempt to establish
   LE connection is made.
2. If the connection is successful, the central initiates disconnect to
   the peripheral and then restarts the scan.
3. If there are no connections, the central keeps scanning continuously.

The sample is used to demonstrate the Central mode capabilities of BLE and
hence a disconnect is issued right after connection with a peripheral,
to keep the central in scan mode.

REQUIREMENTS
************

* BlueZ running on the host, or
* A board with Bluetooth LE support

BUILDING AND RUNNING
********************
Build and flash the sample as follows, replacing board_name with your
target board:

.. code-block:: shell

    west build -b board_name samples/bluetooth/central

>     west flash

Reviewer 3: use .. zephyr-app-commands::

To test Central's scanning functionality, either flash the peripheral
sample on a second compatible board or use an off-the-shelf BLE enabled
device that can act as a peripheral (eg. smartphone, smartwatch, etc.).

ADDING SUPPORT FOR NEW BOARDS:
******************************

To add Bluetooth Central support for a new board:

* Ensure that the following BLE configurations are enabled in prj.conf
.. code-block:: shell

    CONFIG_BT=y
    CONFIG_BT_CENTRAL=y

* CONFIG_BT=y: This enables the Bluetooth stack.

> * CONFIG_BT_CENTRAL=y: This specifically enables the BLE Central role.

Reviewer 3: ```
* Ensure that the following BLE configurations are enabled in prj.conf
  .. code-block:: cfg

     # Enable BLE
     CONFIG_BT=y
     # Enable central role
     CONFIG_BT_CENTRAL=y
```

Reviewer 3: please apply this change as suggested

Author: Done

Author: Done

Author: @kartben 
I have made the changes.Can you please look into it?

to the first one with a strong enough signal.

Core features
*************

Scanning for devices:
=====================

The application initiates a passive scan to detect nearby BLE devices.
It specifically looks for devices that have a signal strength greater
than -50dBm.This threshold helps the app filter out weaker signals,
ensuring it only interacts with devices that are within a reasonable
range for communication.

Connection handling:
====================

1. The central scans for peripheral devices and if it finds a peripheral
   which has a signal strength higher than -50dBm, an attempt to establish
   LE connection is made.
2. If the connection is successful, the central initiates disconnect to
   the peripheral and then restarts the scan.
3. If there are no connections, the central keeps scanning continuously.

The sample is used to demonstrate the Central mode capabilities of BLE and
hence a disconnect is issued right after connection with a peripheral,
to keep the central in scan mode.

Requirements
************

* BlueZ running on the host, or
* A board with Bluetooth LE support

Building and running

> ********************

Reviewer 3: for consistency

```suggestion
********************

```

Author: Done

than -50dBm.This threshold helps the app filter out weaker signals,
ensuring it only interacts with devices that are within a reasonable
range for communication.

Connection handling
===================

1. The central scans for peripheral devices and if it finds a peripheral
   which has a signal strength higher than -50dBm, an attempt to establish
   LE connection is made.
2. If the connection is successful, the central initiates disconnect to
   the peripheral and then restarts the scan.
3. If there are no connections, the central keeps scanning continuously.

The sample is used to demonstrate the Central mode capabilities of BLE and
hence a disconnect is issued right after connection with a peripheral,
to keep the central in scan mode.

Requirements
************

* BlueZ running on the host, or
* A board with Bluetooth LE support

Building and running
********************

Build and flash the sample as follows, replacing board_name with your
target board:

.. zephyr-app-commands::
   :zephyr-app: samples/bluetooth/central
   :board: board_name
   :goals: build flash
   :compact:

To test Central's scanning functionality, either flash the peripheral

> sample on a second compatible board or use an off-the-shelf BLE enabled

Author: done

device that can act as a peripheral (eg. smartphone, smartwatch, etc.).

Adding support for new boards:
******************************

To add Bluetooth Central support for a new board:

* Ensure that the following BLE configurations are enabled in prj.conf

  .. code-block:: cfg

     # Enable BLE
     CONFIG_BT=y
     # Enable central role

>      CONFIG_BT_CENTRAL=y

Author: @kartben I have made the changes.Can you please look into it?

for other Bluetooth LE devices and establishing a connection to the first one with a
strong enough signal.

Core Capabilities
*****************

> * This sample starts passive scanning and finds the devices with rssi greater than -50.

Reviewer 4: `rssi greater than -50` ->  `RSSI greater than -50`
